compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -524,103 +524,98 @@ TR::Register *OMR::X86::TreeEvaluator::tableEvaluator(TR::Node *node, TR::CodeGe
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), result->getLowOrder(), cg); // if (high0 == high1) then t2 = t1 = (low0 < low1)
+
+   generateRegRegInstruction(TEST1RegReg,  node, result->getHighOrder(), result->getHighOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getLowOrder(),  oprand0->getLowOrder(),  cg);
+   generateRegRegInstruction(MOV4RegReg,   node, result->getHighOrder(), oprand0->getHighOrder(), cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getLowOrder(),  oprand1->getLowOrder(),  cg);
+   generateRegRegInstruction(CMOVE4RegReg, node, result->getHighOrder(), oprand1->getHighOrder(), cg);
+   node->setRegister(result);
+   for (int i = 0; i < node->getNumChildren(); i++)
       {
-      generateRegRegInstruction(cmp_op, node, reg, trgReg, cg);
-      generateLabelInstruction(isMin ? JG4 : JL4, node, doneLabel, false, cg);
-      generateRegRegInstruction(move_op, node, trgReg, reg, cg);
-
-      dep = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg->trMemory());
-      dep->addPostCondition(trgReg, TR::RealRegister::NoReg, cg);
-      dep->addPostCondition(reg, TR::RealRegister::NoReg, cg);
+      cg->decReferenceCount(node->getChild(i));
       }
-
-   generateLabelInstruction(LABEL, node, doneLabel, dep, cg);
-
-   node->setRegister(trgReg);
+   return result;
+   }
+TR::Register *OMR::X86::TreeEvaluator::minmaxEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   if (TR::Compiler->target.is32Bit() && 
+       (node->getOpCodeValue() == TR::lmin || node->getOpCodeValue() == TR::lmax))
+      {
+      return minmaxEvaluator64BitPrimitiveOnIA32(node, cg);
+      }
+
+   TR_X86OpCodes CMP  = BADIA32Op;
+   TR_X86OpCodes MOV  = BADIA32Op;
+   TR_X86OpCodes CMOV = BADIA32Op;
+   switch (node->getOpCodeValue())
+      {
+      case TR::imin:
+         CMP = CMP4RegReg;
+         MOV = MOV4RegReg;
+         CMOV = CMOVG4RegReg;
+         break;
+      case TR::imax:
+         CMP = CMP4RegReg;
+         MOV = MOV4RegReg;
+         CMOV = CMOVL4RegReg;
+         break;
+      case TR::lmin:
+         CMP = CMP8RegReg;
+         MOV = MOV8RegReg;
+         CMOV = CMOVG8RegReg;
+         break;
+      case TR::lmax:
+         CMP = CMP8RegReg;
+         MOV = MOV8RegReg;
+         CMOV = CMOVL8RegReg;
+         break;
+      default:
+         TR_ASSERT(false, "INCORRECT IL OPCODE.");
+         break;
+      }
+   
+   auto oprand0 = cg->evaluate(node->getChild(0));
+   auto oprand1 = cg->evaluate(node->getChild(1));
+   auto result = cg->allocateRegister();
+   generateRegRegInstruction(CMP,  node, oprand0, oprand1, cg);
+   generateRegRegInstruction(MOV,  node, result,  oprand0, cg);
+   generateRegRegInstruction(CMOV, node, result,  oprand1, cg);
+
+   node->setRegister(result);
    for (int i = 0; i < node->getNumChildren(); i++)
       {
       cg->decReferenceCount(node->getChild(i));
       }
```

Comment:
Is a loop necessary?  Can `min` or `max` ever have anything other than 2 children?  I'd say just do two `decReferenceCount`s to make the code easier to read.

Reply:
Sure, the ILOp should only have 2 children. I've updated this part to two `decReferenceCount`